core/app/helpers/spree/base_helper.rb

```diff
@@ -25,6 +25,24 @@ def display_price(product_or_variant)
         to_html
     end
 
+    def localized_item_link(item, locale)
+      return if item.destination.nil?
+
+      output_locale = if locale == current_store.default_locale
```

Comment:
@MatthewKennedy you can use `locale_param` for this - https://github.com/spree/spree/blob/c47e591882d79e766847c627773d29186aa2224b/core/lib/spree/core/controller_helpers/locale.rb#L50